include/pops/network.hpp

```diff
@@ -544,6 +664,68 @@ class Network
         }
     }
 
+    /**
+     * @brief Convert string to cost
+     *
+     * @param text String with cost
+     * @return Cost as number
+     */
+    static double cost_from_text(const std::string& text)
+    {
+        try {
+            return std::stod(text);
+        }
+        catch (const std::invalid_argument& err) {
+            if (string_contains(text, '"') || string_contains(text, '\'')) {
+                throw std::invalid_argument(
+                    std::string("Text for cost cannot contain quotes "
+                                "(only digits are allowed): ")
+                    + text);
+            }
+            if (text.empty()) {
+                throw std::invalid_argument("Text for cost cannot be an empty string");
+            }
+            else {
+                throw std::invalid_argument(
+                    std::string("Text cannot be converted to cost "
+                                "(only digits are allowed): ")
+                    + text);
+            }
+        }
+        catch (const std::out_of_range& err) {
+            throw std::out_of_range(
+                std::string("Numerical value too large for cost: ") + text);
+        }
+    }
+
+    template<typename InputStream>
+    static bool stream_has_cost_column(InputStream& stream, char delimeter)
+    {
+        // Get header to determine what is included.
+        auto starting_position = stream.tellg();
+        std::string line;
+        std::getline(stream, line);
+        std::istringstream line_stream{line};
+        std::string label;
+        int column_number = 0;
+        while (std::getline(line_stream, label, delimeter)) {
+            column_number++;
+            if (column_number == 1 && label != "node_1") {
+                // The right label is not there. Assuming it is without a header.
+                stream.seekg(starting_position);
+                break;
+            }
+            if (label == "cost") {
```

Comment:
Shouldn't the header be documented somewhere?

Reply:
Documented in the load method description where the rest of the format is documented.